### DISCUSSION TITLE
Schedule Helper Feature Requests - UI Granularity, Reloading

### DISCUSSION THREAD

**Discussion by SirNate0** [home-assistant/frontend#14825] [Other feature requests]
As is, creating a schedule with the UI is not particularly usable for anything other than very broad schedules that match half-hour intervals (HH::00 or HH:30). Internally, though, the schedules are saved with at least second granularity. I was attempting to create a schedule that needs to start at 40 after, but there was no way to do this from the UI.

Instead, I had two options:

1. (Re)Create the schedule in YAML. This was more frustrating than necessary, as the layout of the schedule in the `.storage/schedule` file is json with a slightly different format than the YAML (shown at <https://www.home-assistant.io/integrations/schedule/>). Since JSON is generally (always?) valid YAML, simply running it through a site like yamllint.com is enough to do most of the reformatting, though some manual changes were still required. **Unfortunately** with this option the schedule cannot be edited in the UI, and nor is it even displayed read-only.
2. Edit the (seemingly undocumented) file `.storage/schedule` with the times that I would prefer rather than the half-hour increments. **However** there seems to be no way to force a reload of this file. While `Schedule: Reload` sounds like it should reload it, it does not. Same for the `Developer Tools > YAML > YAML configuration reloading > SCHEDULE` button. Seemingly the only way to get it to reload the file is to restart home assistant.

---

So, the feature requests:

1. Allow finer control of schedule event timing in the UI. At least 5 minute granularity, I would say. Either directly from the week view or as an edit option once a calendar event is selected (similar to how Google calendars works). Alternatively/Additionally, allow editing the UI schedules as YAML, similarly to how Automations can be edited as YAML or in the UI.
2. Provide a way to reload the `.storage/schedule` file as well, e.g. through the `Schedule: Reload` service.
3. Display the read-only YAML schedules in the UI, even if they are not editable.

**Comment by bernieke** [comment]
Apparently granularity has been implemented already.
